scala-package/examples/src/main/scala/ml/dmlc/mxnet/examples/module/MnistMlp.scala

```diff
@@ -0,0 +1,158 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ml.dmlc.mxnet.examples.module
+
+import ml.dmlc.mxnet._
+import ml.dmlc.mxnet.module.Module
+import ml.dmlc.mxnet.DataDesc._
+import ml.dmlc.mxnet.optimizer.SGD
+import org.kohsuke.args4j.{Option, CmdLineParser}
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+
+object MnistMlp {
+  private val logger = LoggerFactory.getLogger(classOf[MnistMlp])
+
+  def getSymbol: Symbol = {
+    val data = Symbol.Variable("data")
+    val fc1 = Symbol.FullyConnected(name = "fc1")(data)(Map("num_hidden" -> 128))
+    val act1 = Symbol.Activation(name = "relu1")(fc1)(Map("act_type" -> "relu"))
+    val fc2 = Symbol.FullyConnected(name = "fc2")(act1)(Map("num_hidden" -> 64))
+    val act2 = Symbol.Activation(name = "relu2")(fc2)(Map("act_type" -> "relu"))
+    val fc3 = Symbol.FullyConnected(name = "fc3")(act2)(Map("num_hidden" -> 10))
+    val softmax = Symbol.SoftmaxOutput(name = "softmax")(fc3)()
+    softmax
+  }
+
+  def runIntermediateLevelApi(train: DataIter, eval: DataIter, cmdLine: MnistMlp): Unit = {
+    // Intermediate-level API
+    val mod = new Module(getSymbol)
+    mod.bind(dataShapes = train.provideData, labelShapes = Some(train.provideLabel))
+    mod.initParams()
+    mod.initOptimizer(optimizer = new SGD(learningRate = 0.01f, momentum = 0.9f))
+
+    val metric = new Accuracy()
+
+    for (epoch <- 0 until cmdLine.numEpoch) {
+      while (train.hasNext) {
+        val batch = train.next()
+        mod.forward(batch)
+        mod.updateMetric(metric, batch.label)
+        mod.backward()
+        mod.update()
+      }
+
+      val (name, value) = metric.get
+      logger.info(s"epoch $epoch $name=$value")
+      metric.reset()
+      train.reset()
+    }
+
+    // High-level API
+    train.reset()
+  }
+
+  def runHighLevelApi(train: DataIter, test: DataIter, cmdLine: MnistMlp): Unit = {
+    val mod = new Module(getSymbol)
+    mod.fit(train, evalData = scala.Option(test), numEpoch = cmdLine.numEpoch)
+
+    // prediction iterator API
+    var iBatch = 0
+    test.reset()
+    while (test.hasNext) {
+      val batch = test.next()
+      val preds = mod.predict(batch)
+      val predLabel: Array[Int] = NDArray.argmax_channel(preds(0)).toArray.map(_.toInt)
+      val label = batch.label(0).toArray.map(_.toInt)
+      val acc = (predLabel zip label).map { case (py, y) =>
```

Comment:
most of places in Spark uses method style....but there are also some places in Spark SQL to have inconsistent usage...so....no solution there...